### PR TITLE
Nano: Add key feature and how to guide for multi-instance inference

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Add key feature and how to guide for multi-instance inference

### 1. Why the change?

Provide guides for users to let them know how to use multi-instance inference.

### 2. User API changes

N/A

### 3. Summary of the change 

Add key feature and how to guide for multi-instance inference

### 4. How to test?
- [ ] Unit test
- [ ] Document test
    key feature:
    <https://yishuotestdoc.readthedocs.io/en/add-ulti-instance-doc/doc/Nano/Overview/pytorch_inference.html#multi-instance-acceleration>
    how to guide:
    <https://yishuotestdoc.readthedocs.io/en/add-ulti-instance-doc/doc/Nano/Howto/Inference/PyTorch/multi_instance_pytorch_inference.html>

### CODE REVIEW

**Comment by MeouSker77** [comment]
@Siritao @ACupofAir please also take a look

**Review by Oscilloscope98 — COMMENTED** [review]
Could you also correct the api doc syntax for `to_multi_instance` at the meantime in this PR?

image

You would only need to add one empty line under "Transform a model to multi-instance inference model.", like below:
```python
    @staticmethod
    def to_multi_instance(model: nn.Module, num_processes: int = 4,
                          cores_per_process: int = None,
                          cpu_for_each_process: List[List] = None) -> _MultiInstanceModel:
        """
        Transform a model to multi-instance inference model.
                                                                
        :param model: The model to transform.
        :param num_processes: The number of processes to use, default to 4.
        :param cores_per_process: Number of CPU cores used by each process,
            default to `None`, means decided automatically.
        :param cpu_for_each_process: Specify the CPU cores used by each process,
            default to `None`, if set, it will override `num_processes` and `cores_per_process`.
        :return: Model with multi-instance inference acceleration.
        """
```
Others LGTM.

File: python/nano/tutorial/notebook/inference/pytorch/multi_instance_pytorch_inference.ipynb [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    "# each process will use 2 CPU cores\n",
+    "multi_model = InferenceOptimizer.to_multi_instance(model_ft, num_processes=4, cores_per_process=2)\n",
+    "y_hat_list = multi_model(batch_list)\n",
+    "\n",
+    "# Use 4 processes to run inference,\n",
+    "# the first process will use core 0, the second process will use core 1,\n",
+    "# the third process will use core 2 and 3, the fourth process will use core 4 and 5\n",
+    "multi_model = InferenceOptimizer.to_multi_instance(model_ft, cpu_for_each_process=[[0], [1], [2,3], [4,5]])\n",
+    "y_hat_list = multi_model(batch_list)"
+   ]
```
[the remark this answers is not in the document]
Comment (MeouSker77) [reply python/nano/tutorial/notebook/inference/pytorch/multi_instance_pytorch_inference.ipynb:175]: > Would it be better to add a note box here to explain the case if cores_per_process, cpu_for_each_process, and num_processe are used together. Do they have some overriding logic?

Added explanation, `cpu_for_each_process` will override `num_processes` and `cores_per_process`

**Comment by MeouSker77** [comment]
> Could you also correct the api doc syntax for to_multi_instance at the meantime in this PR?
done

**Review by rnwang04 — APPROVED** [review]

**Pull request merged**